### ISSUE TITLE
[lesion_detector_3DCE] resnet

### ISSUE THREAD

**Issue by pengchengu** [rsummers11/CADLab#7]
I'm wondering that have you ever tried to change the backbone from vgg16 to resnet50 or resnet101. I know it seems difficult to change to resnet, as it will decrease the resolution of the feature map with RPN_FEAT_STRIDE set to 16 and if closing the first pooling layer of resnet it will increase the computation cost a lot. But resnet has been proved outperforming vgg in many detection cases.
Do you have any opinion on this?

**Comment by viggin** [comment]
Thank you for your suggestion. My preliminary results show that VGG-16 = truncated DenseNet (conv1-3) > ResNet-18 > AlexNet > ResNet-50. However, this can also be due to the imperfect training settings, as I didn't spend too much time to tune the other backbones. I believe that some latest backbones may definitely work better.

**Comment by maggiezha** [comment]
I am trying to run test.sh or train.sh of lesion_detect_3DCE, and got a problem:
I upzip the DeepLesion dataset, so the data is at: /home/data/DeepLesion/Images_png/000001_01_01/, etc,
and I modified default.yml to be:
dataset_path: '/home/data/DeepLesion/'
image_path: '/home/data/DeepLesion/Images_png/'
Then after running test.sh, it says:
INFO:root:Loading images with 2 threads.
INFO:root:loading parameters from model/3DCE 1 image 3 slice-0000.params
[16:12:54] include/dmlc/logging.h:308: [16:12:54] src/io/local_filesys.cc:166: Check failed: allow_null LocalFileSystem: fail to open "model/3DCE 1 image 3 slice-0000.params"
Also if I print out 'mask' after line 188 of load_ct_img.py, it shows:
array([[False, False, False, ..., False, False, False],
[False, False, False, ..., False, False, False],
[False, False, False, ..., False, False, False],
...,
[False, False, False, ..., False, False, False],
[False, False, False, ..., False, False, False],
[False, False, False, ..., False, False, False]]))
Could you please help me understand what and where should I have for 'model/3DCE 1 image 3 slice-0000.params' and how could I load the dataset properly? Many thanks!

**Comment by pengchengu** [comment]
> I am trying to run test.sh or train.sh of lesion_detect_3DCE, and got a problem:
> I upzip the DeepLesion dataset, so the data is at: /home/data/DeepLesion/Images_png/000001_01_01/, etc,
> and I modified default.yml to be:
> dataset_path: '/home/data/DeepLesion/'
> image_path: '/home/data/DeepLesion/Images_png/'
> Then after running test.sh, it says:
> INFO:root:Loading images with 2 threads.
> INFO:root:loading parameters from model/3DCE 1 image 3 slice-0000.params
> [16:12:54] include/dmlc/logging.h:308: [16:12:54] src/io/local_filesys.cc:166: Check failed: allow_null LocalFileSystem: fail to open "model/3DCE 1 image 3 slice-0000.params"
> Also if I print out 'mask' after line 188 of load_ct_img.py, it shows:
> array([[False, False, False, ..., False, False, False],
> [False, False, False, ..., False, False, False],
> [False, False, False, ..., False, False, False],
> ...,
> [False, False, False, ..., False, False, False],
> [False, False, False, ..., False, False, False],
> [False, False, False, ..., False, False, False]]))
> Could you please help me understand what and where should I have for 'model/3DCE 1 image 3 slice-0000.params' and how could I load the dataset properly? Many thanks!
Hi maggiezha,
The dataset setting is correct, the problem i believe is due to you tested the model but you actually hadn't train it before , so you have no model for testing.
First, you should download the Deeplesion dataset, unzip it and modify the defalut.yml, which you have already done. Then run the train.sh to train you model.  When the training finished, modify the setting 'begin_epoch: ' in default.yaml to the number of epoch you have trained, then run the test.sh, which should works.

**Comment by maggiezha** [comment]
Thank you very much for the help @pengchengu and @viggin , the problem of loading the data is gone now. I still have some questions to run it:
1. @viggin I guess you meant to set  'validate_at_begin' to TRUE?
2. I am wondering where are the data stored after using load_ct_img.py to preprocess the dataset? As I also wanna use the data after preprocessing to test my own code
3. When I tried to run 'train.sh', it says 'INFO: root: loading parameters from /home/yk/ct/data/imagenet_models/MXNet/vgg16-0000.params', so I assume I should download the pretrained model 'vgg16-0000.params' and modify the path to where I store it, but I couldn't find where to modify the path in the code, could you please let me know where should I change? Thank you for the great work @viggin

**Comment by maggiezha** [comment]
It is working well, thank you very much for the help @viggin  @pengchengu
